readthedocs/restapi/utils.py

```diff
@@ -44,23 +69,67 @@ def sync_versions(project, versions, type):  # pylint: disable=redefined-builtin
 
                 log.info(
                     '(Sync Versions) Updated Version: [%s=%s] ',
-                    version['verbose_name'],
-                    version['identifier'],
+                    version_name,
+                    version_id,
                 )
         else:
             # New Version
             created_version = Version.objects.create(
                 project=project,
                 type=type,
-                identifier=version['identifier'],
-                verbose_name=version['verbose_name'],
+                identifier=version_id,
+                verbose_name=version_name,
             )
             added.add(created_version.slug)
+    if not has_user_stable:
+        stable_version = (
+            project.versions
+            .filter(slug=STABLE, type=type)
+            .first()
+        )
+        if stable_version:
+            # Put back the RTD's stable version
+            stable_version.machine = True
+            stable_version.save()
+    if not has_user_latest:
+        latest_version = (
+            project.versions
+            .filter(slug=LATEST, type=type)
+            .first()
+        )
+        if latest_version:
+            # Put back the RTD's latest version
+            latest_version.machine = True
+            latest_version.identifier = project.get_default_branch()
+            latest_version.verbose_name = LATEST_VERBOSE_NAME
+            latest_version.save()
     if added:
         log.info('(Sync Versions) Added Versions: [%s] ', ' '.join(added))
     return added
 
 
+def set_or_create_version(project, slug, version_id, verbose_name, type_):
+    version = (
+        project.versions
+        .filter(slug=slug)
+        .first()
+    )
+    if version:
+        version.identifier = version_id
+        version.machine = False
+        version.type = type_
+        version.save()
+    else:
+        created_version = Version.objects.create(
+            project=project,
+            type=type_,
+            identifier=version_id,
+            verbose_name=verbose_name,
+        )
+        return created_version
+    return None
```

Comment:
There is a common behaviour from different Django methods where something like this returns a tuple: `(object, created)` which is easier to read and also the _updated/set_ object is returned in case it's needed.

Reply:
I think I got it, I'll see how this can be refactored